src/components/events/CheckInModal/index.tsx

```diff
@@ -0,0 +1,62 @@
+import { Modal, Typography } from '@/components/common';
+import { PublicEvent } from '@/lib/types/apiResponses';
+import Image from 'next/image';
+import style from './style.module.scss';
+
+interface CheckInModalProps {
+  open: boolean;
+  event?: PublicEvent;
+  onClose: () => void;
+}
+
+const CHECKIN_TITLES = [
+  'Yippee!',
+  'Brilliant!',
+  'Wahoo!',
+  'Awesome!',
+  'Nice to see you!',
+  'Congrats!',
+  'Hello World!',
+  'Welcome!',
+];
+
+const CheckInModal = ({ open, event, onClose }: CheckInModalProps) => {
+  if (!event) {
+    return null;
+  }
+
+  const { pointValue, title } = event;
+
+  return (
+    <Modal open={open} onClose={onClose}>
+      <div className={style.container}>
+        <div className={style.graphic}>
+          <Image src="/assets/graphics/portal/treasure.svg" alt="treasure" fill />
+        </div>
+        <div className={style.header}>
+          <Typography
+            variant="display/light/small"
+            className={style.headerText}
+            suppressHydrationWarning
+          >
+            {CHECKIN_TITLES[Math.floor(Math.random() * CHECKIN_TITLES.length)]}
+          </Typography>
```

Comment:
You should probably store the random check-in title in a `useMemo`'d variable to avoid the hydration error and the text changing when there's a rerender caused by some other part of the page

Could also have a util function for choosing a random item in an array